### PR TITLE
Request level mode setup

### PR DESCRIPTION
**Pull request by sam110**

### CODE REVIEW

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java [hunk 1]
```diff
@@ -98,13 +117,36 @@ public static void setTestDataId(String testDataId) {
...
   static synchronized void setMode(Mode mode) {
     globalProfileSetting.setMode(mode);
   }
 
+  static synchronized void setRequestMode(String modeHeader) {
+    ProfileSetting profileSetting = new ProfileSetting();
+
+    if (Mode.DYNAMIC.equals(globalProfileSetting.getMode())) {
+      try {
+        if (StringUtils.isEmpty(modeHeader) || Mode.DYNAMIC.name().equals(modeHeader)) {
```
Comment (parthshyara) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java:135]: use null check and String.isEmpty() as StringUtils is not extensively required here. And commons-lang3 dependecy can be removed.

Comment (sam110) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java]: Done

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java [hunk 2]
```diff
@@ -56,6 +56,20 @@ protected String initialValue() {
         }
       };
   private static ProfileSetting globalProfileSetting = new ProfileSetting();
+
+  private static InheritableThreadLocal<ProfileSetting> threadLocalProfileSetting =
```
Comment (saxenaluv) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java:60]: variable name can be renamed to requestProfileSetting

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java [hunk 3]
```diff
@@ -81,6 +95,10 @@ public static void clearGlobalPerRequestID() {
     GLOBAL_PER_REQUEST_ID.remove();
   }
 
+  public static void clearThreadLocalProfileSetting() {
```
Comment (saxenaluv) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java:98]: method name can be renamed to requestProfileSetting.

Comment (sam110) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java]: Done

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java (hunk 2 above)
Comment (sam110) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java]: Done

**Comment by arunachalams** [comment]
I have reverted the changes.

**Pull request merged**